web/src/components/Upload/util.ts

```diff
@@ -1,43 +1,132 @@
+  try {
+    const endDatetime: moment.Moment = startDatetime.clone()
+    const testLength: string = textLines[8].split(',')[1]
+    const [days, hours, minutes]: Array<string> = testLength.split(':')
+    endDatetime
+      .add(days, 'days')
+      .add(hours, 'hours')
+      .add(minutes, 'minutes')
+    return endDatetime
+  } catch(e){
+    if (e instanceof TypeError){
+      console.warn(e)
+      return moment('')
+    } else{
+      throw e
+    }
   }
+}
 
-  if (files.length > 2) {
-    return 'We need exactly one GPS log file and one DusTrak cvs file. Please remove additional files.'
+export const validateFiles = async (
+  files: Array<FileWithPath>
+): Promise<ValidateMeta> => {
+  const dustrakData: DustrakMeta = {
+    startDatetime: '',
+    endDatetime: ''
+  }
+  const validateData: ValidateMeta = {
+    message: '',
+    dustrakMeta: dustrakData
   }
 
-  if (files.length === 2) {
-    let logFile: FileWithPath | undefined
-    let csvFile: FileWithPath | undefined
+  if (files.length > 0 && files.length < 2) {
+    validateData.message =
+      'We need one GPS log file and one DusTrak cvs file. Please add a file to continue.'
+  } else if (files.length > 2) {
+    validateData.message =
+      'We need exactly one GPS log file and one DusTrak cvs file. Please remove additional files.'
+  } else if (files.length === 2) {
+    let logFile: File | undefined
+    let csvFile: File | undefined
 
     files.forEach(file => {
-      if (file.name.endsWith('.csv')) {
+      if (file.type === 'text/csv') {
         csvFile = file
       }
 
-      if (file.name.endsWith('.log')) {
+      if (file.type === 'text/x-log') {
         logFile = file
       }
     })
 
     if (!logFile || !csvFile) {
-      return 'We need one GPS log file and one DusTrak csv file. Please replace one of your files to continue.'
+      validateData.message =
+        'We need one GPS log file and one DusTrak csv file. Please replace one of your files to continue.'
+    } else {
+      const [logText, csvText]: Array<string> = await Promise.all([
+        logFile.text(),
+        csvFile.text()
+      ])
+      const csvTextLines: Array<string> = csvText.split('\n', 9)
+      const dustrakStart: moment.Moment = getDustrakStart(csvTextLines)
+      const dustrakEnd: moment.Moment = getDustrakEnd(
+        csvTextLines,
+        dustrakStart
+      )
+      const logTextLines: Array<string> = logText.split('\n', 10)
+      const gpsStart: moment.Moment = getGpsStart(logTextLines)
+      if (
+        !dustrakStart.isValid() ||
+        !dustrakEnd.isValid() ||
+        !gpsStart.isValid()
+      ) {
+        validateData.message =
+          'Files could not be uploaded. Try again or choose a different file.'
+      } else if (
+        dustrakStart.subtract(2, 'minutes') <= gpsStart &&
+        gpsStart <= dustrakStart.add(2, 'minutes')
+      ) {
+        validateData.message = ''
+        dustrakData.startDatetime = dustrakStart.format()
```

Comment:
from an efficiency standpoint, it makes sense to just grab the dustrakData while we're validating it, but I think since this function's main purpose is to just validate, and it's already doing a lot, we might want to do this in a separate function, even if it comes at the expense of needing to call that parse function again.

Reply:
**Response**
To implement this, I refactored the `csvFile` and `logFile` assignment step into a dedicated function- `identFiles`

**Context**
Trying to stay DRY when finding the csv file again in the data assignment step